### PR TITLE
Proxy settings have been implemented (closes #1206)

### LINKED ISSUE #1206: Implement proxy settings for TestCafe
### Are you requesting a feature or reporting a bug?

feature

### What is the current behavior?

In corporate networks, where browser have specified proxy settings for access to WEB, TestCafe cannot establish connection to the remote resources, cause all requests are sent directly to the remote host, bypassing corporate proxy.

### What is the expected behavior?

TestCafe should provide capability to setup proxy options, like `http proxy url`, `http proxy port` etc

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @inikulin, @AlexanderMoskovkin

### CODE REVIEW

File: test/functional/site/anonym-proxy.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+var http   = require('http');
```
Comment (georgiy-abbasov) [inline test/functional/site/anonym-proxy.js]: Maybe it should be called "trusted-proxy" ?

File: test/functional/fixtures/api/es-next/proxy/test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+var os = require('os');
+
+const ANONYM_PROXY_URL      = os.hostname() + ':3004';
+const TRANSPARENT_PROXY_URL = os.hostname() + ':3005';
+
+it('Should use proxy to open page', function () {
```
Comment (georgiy-abbasov) [inline test/functional/fixtures/api/es-next/proxy/test.js]: Missed `describe` annotation

File: src/runner/index.js [hunk 3]
```diff
@@ -19,11 +19,12 @@ export default class Runner extends EventEmitter {
     constructor (proxy, browserConnectionGateway) {
         super();
 
-        this.proxy               = proxy;
+        this.hammerhead          = proxy;
```
Comment (inikulin) [inline src/runner/index.js]: why rename?

Comment (inikulin) [inline src/runner/index.js]: Keep `proxy`

File: test/functional/fixtures/api/es-next/proxy/test.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+var os = require('os');
+
+const TRUSTED_PROXY_URL     = os.hostname() + ':3004';
+const TRANSPARENT_PROXY_URL = os.hostname() + ':3005';
+
+describe('[API] Using external proxy', function () {
```
Comment (inikulin) [inline test/functional/fixtures/api/es-next/proxy/test.js]: It has nothing to do with test API and shouldn't be in this directory

File: src/runner/index.js (hunk 3 above)
Comment (georgiy-abbasov) [inline src/runner/index.js]: Ivan's remark:
method should be called `useProxy`

```javascript
runner.useProxy()
```

File: src/cli/argument-parser.js [hunk 5]
```diff
@@ -106,6 +106,7 @@ export default class CLIArgumentParser {
             .option('--speed <factor>', 'set the speed of test execution (0.01 ... 1)')
             .option('--ports <port1,port2>', 'specify custom port numbers')
             .option('--hostname <name>', 'specify the hostname')
+            .option('--proxy-url <proxyUrl>', 'specify proxy url')
```
Comment (VasilyStrelyaev) [inline src/cli/argument-parser.js]: we now call this

```
            .option('--proxy <hostname>', 'specify the hostname of the proxy server')
```

File: src/cli/argument-parser.js [hunk 6]
```diff
@@ -106,6 +106,7 @@ export default class CLIArgumentParser {
             .option('--speed <factor>', 'set the speed of test execution (0.01 ... 1)')
             .option('--ports <port1,port2>', 'specify custom port numbers')
             .option('--hostname <name>', 'specify the hostname')
+            .option('--proxy <hostname>', 'specify the hostname of the proxy server')
```
Comment (inikulin) [inline src/cli/argument-parser.js]: It's not a `hostname`, it's `host`, since it's can contain port as well.

File: src/runner/index.js [hunk 7]
```diff
@@ -152,6 +153,12 @@ export default class Runner extends EventEmitter {
         return this;
     }
 
+    useProxy (externalProxyUrl) {
```
Comment (inikulin) [inline src/runner/index.js]: It's `host` not URL

**Pull request merged**